From 55096239fda3124a48002252921c0bf3fe663499 Mon Sep 17 00:00:00 2001
From: Zhang Xiaoxu <zhangxiaoxu5@huawei.com>
Date: Wed, 7 Dec 2022 18:39:35 +0800
Subject: [PATCH] dm btree spine: show warning if node_check failed in
 node_prep_for_write()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 55096239fda3124a48002252921c0bf3fe663499
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5JCAH
CVE: NA

--------------------------------

The BUG_ON is unneed Since f73e2e70ec48 ("dm btree spine: remove
paranoid node_check call in node_prep_for_write()") merged in v5.13.

For debug reason, we also want to know the data on disk is corrupted
by write or disk fault. So also add check and print some info when
data corrupted.

Signed-off-by: Zhang Xiaoxu <zhangxiaoxu5@huawei.com>
Reviewed-by: Hou Tao <houtao1@huawei.com>
Reviewed-by: Jason Yan <yanaijie@huawei.com>
Signed-off-by: Yongqiang Liu <liuyongqiang13@huawei.com>
Reviewed-by: Zhang Xiaoxu <zhangxiaoxu5@huawei.com>
Reviewed-by: Hou Tao <houtao1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/md/persistent-data/dm-btree-spine.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/md/persistent-data/dm-btree-spine.c b/drivers/md/persistent-data/dm-btree-spine.c
index c4b386b2be97..859aae0a8683 100644
--- a/drivers/md/persistent-data/dm-btree-spine.c
+++ b/drivers/md/persistent-data/dm-btree-spine.c
@@ -30,6 +30,8 @@ static void node_prepare_for_write(struct dm_block_validator *v,
 	h->csum = cpu_to_le32(dm_bm_checksum(&h->flags,
 					     block_size - sizeof(__le32),
 					     BTREE_CSUM_XOR));
+	if (node_check(v, b, 4096))
+		DMWARN_LIMIT("%s node_check failed", __func__);
 }
 
 static int node_check(struct dm_block_validator *v,
-- 
2.33.0

